### PR TITLE
Add mathcomp-analysis to the bench.

### PR DESCRIPTION
**Pull request by ppedrot**
This is a better end-user of some mathcomp-y features than the core mathcomp packages.

### CODE REVIEW

File: dev/bench/bench.sh [hunk 1]
```diff
@@ -80,7 +80,7 @@ check_variable () {
...
+: "${coq_opam_packages:=coq-test-suite coq-bignums coq-hott coq-performance-tests-lite coq-engine-bench-lite coq-mathcomp-ssreflect coq-mathcomp-fingroup coq-mathcomp-algebra coq-mathcomp-solvable coq-mathcomp-field coq-mathcomp-character coq-mathcomp-odd-order coq-mathcomp-analysis coq-math-classes coq-corn coq-compcert coq-equations coq-metacoq-utils coq-metacoq-common coq-metacoq-template coq-metacoq-pcuic coq-metacoq-safechecker coq-metacoq-erasure coq-metacoq-translations coq-color coq-coqprime coq-coqutil coq-bedrock2 coq-rewriter coq-fiat-core coq-fiat-parsers coq-fiat-crypto-with-bedrock coq-unimath coq-coquelicot coq-iris-examples coq-verdi coq-verdi-raft coq-fourcolor coq-rewriter-perf-SuperFast coq-vst coq-category-theory coq-neural-net-interp-computed-lite}"
```
Comment (SkySkimmer) [inline dev/bench/bench.sh:83]: should we add the remaining dependencies of classical to the list too? I believe that would be `coq-mathcomp-finmap coq-mathcomp-classical coq-mathcomp-bigenough`
bigenough is rather trivial (<1s in https://github.com/coq/coq/pull/19400/#issuecomment-2247334026)
finmap was in the 10s which is also pretty small
however classical is in the 1min range and showed -17% in the above linked bench so it seems to be doing some work

Comment (ppedrot) [reply dev/bench/bench.sh:83]: I'm not sure it's worth it, mathcomp-analysis should subsume the coverage of these devs, and a 1-minute dev is still too sensitive to noise in my opinion.

**Review by SkySkimmer — APPROVED** [review]
I think adding classical still could be worthwhile but let's get on with it

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Pull request merged**